src/hotspot/share/oops/compressedKlass.cpp

```diff
@@ -71,25 +75,39 @@ void CompressedKlassPointers::initialize(address addr, size_t len) {
     //  encoding. We also set the expected value range to 4G (encoding range
     //  cannot be larger than that).
 
-    _base = addr;
+    thebase = addr;
 
   } else {
 
     // (Note that this case is almost not worth optimizing for. CDS is typically on.)
     if ((addr + len) <= (address)KlassEncodingMetaspaceMax) {
-      _base = 0;
+      thebase = nullptr;
     } else {
-      _base = addr;
+      thebase = addr;
     }
   }
 
-  assert(is_valid_base(_base), "Address " PTR_FORMAT " was chosen as encoding base for range ["
-                               PTR_FORMAT ", " PTR_FORMAT ") but is not a valid encoding base",
-                               p2i(_base), p2i(addr), p2i(addr + len));
+  assert(is_valid_base(thebase), "Address " PTR_FORMAT " was chosen as encoding base for range ["
+                              PTR_FORMAT ", " PTR_FORMAT ") but is not a valid encoding base",
+                              p2i(thebase), p2i(addr), p2i(addr + len));
 
   // For SA
+  _base_copy = thebase;
   _shift_copy = LogKlassAlignmentInBytes;
 
+  assert(LogKlassAlignmentInBytes < (1 << encodingShiftWidth), "Shift too large");
+  assert((((uintptr_t)thebase) & ~baseAddressMask) == 0, "Base address " PTR_FORMAT " unaligned", p2i(thebase));
+
+  _value = (UseCompactObjectHeaders ? ((uintptr_t)1 << useCompactObjectHeadersShift) : 0) |
```

Comment:
I am not sure that _value is a good name. Maybe _config or so?

Reply:
Okay, _config it is.